MSAL/test/unit/native_auth/public/delegate/sign_up/SignUpPasswordStartDelegateDispatcherTests.swift

```diff
@@ -0,0 +1,150 @@
+
+final class SignUpPasswordStartDelegateDispatcherTests: XCTestCase {
+
+    private var telemetryExp: XCTestExpectation!
+    private var delegateExp: XCTestExpectation!
+    private var sut: SignUpPasswordStartDelegateDispatcher!
+    private let controllerFactoryMock = MSALNativeAuthControllerFactoryMock()
+    private let correlationId = UUID()
+
+    override func setUp() {
+        super.setUp()
+        telemetryExp = expectation(description: "delegateDispatcher telemetry exp")
+        delegateExp = expectation(description: "delegateDispatcher delegate exp")
+    }
+
+    func test_dispatchSignUpCodeRequired_whenDelegateMethodsAreImplemented() async {
+        let delegate = SignUpPasswordStartDelegateSpy(expectation: delegateExp)
+
+        sut = .init(delegate: delegate, telemetryUpdate: { result in
+            guard case .success = result else {
+                return XCTFail("wrong result")
+            }
+            self.telemetryExp.fulfill()
+        })
+
+        let expectedState = SignUpCodeRequiredState(controller: controllerFactoryMock.signUpController, username: "", flowToken: "flowToken", correlationId: correlationId)
+        let expectedSentTo = "user@contoso.com"
+        let expectedChannelTargetType = MSALNativeAuthChannelType.email
+        let expectedCodeLength = 4
+
+        await sut.dispatchSignUpCodeRequired(
+            newState: expectedState,
+            sentTo: expectedSentTo,
+            channelTargetType: expectedChannelTargetType,
+            codeLength: expectedCodeLength
+        )
+
+        await fulfillment(of: [telemetryExp, delegateExp])
+
+        XCTAssertEqual(delegate.newState, expectedState)
+        XCTAssertEqual(delegate.sentTo, expectedSentTo)
+        XCTAssertEqual(delegate.channelTargetType, expectedChannelTargetType)
+        XCTAssertEqual(delegate.codeLength, expectedCodeLength)
+    }
+
+    func test_dispatchSignUpCodeRequired_whenDelegateOptionalMethodsNotImplemented() async {
+        let delegate = SignUpPasswordStartDelegateOptionalMethodsNotImplemented(expectation: delegateExp)
+        let expectedError = SignUpPasswordStartError(type: .generalError, message: String(format: MSALNativeAuthErrorMessage.requiredDelegateMethod, "onSignUpCodeRequired"))
+
+        sut = .init(delegate: delegate, telemetryUpdate: { result in
+            guard case let .failure(error) = result, let customError = error as? SignUpPasswordStartError else {
+                return XCTFail("wrong result")
+            }
+
+            checkError(customError)
+            self.telemetryExp.fulfill()
+        })
+
+        let expectedState = SignUpCodeRequiredState(controller: controllerFactoryMock.signUpController, username: "", flowToken: "flowToken", correlationId: correlationId)
+        let expectedSentTo = "user@contoso.com"
+        let expectedChannelTargetType = MSALNativeAuthChannelType.email
+        let expectedCodeLength = 4
+
+        await sut.dispatchSignUpCodeRequired(
```

Comment:
Is this call correctly being made when the tests are for PasswordStart?
Also the unit test name test_dispatchSignUpCodeRequired_whenDelegateOptionalMethodsNotImplemented is it correct?

Reply:
Changed (same as [here](https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/1914#discussion_r1404549624))